pallets/clusters/src/lib.rs

```diff
@@ -269,23 +279,21 @@ pub mod pallet {
 			let role_hash = blake2_128(&[&cluster_id.as_slice(), role_name.as_slice()].concat());
 
 			// Check that the caller is the owner of the cluster
-			let cluster = <Clusters<T>>::get(&cluster_id).ok_or(Error::<T>::ClusterNotFound)?;
-			ensure!(who == cluster.owner, Error::<T>::NoPermission);
+			ensure!(who == <Clusters<T>>::get(&cluster_id).ok_or(Error::<T>::ClusterNotFound)?.owner, Error::<T>::NoPermission);
 
 			// At creation there are no Members and no Rules assigned to a Role
 			let new_role = Role { name: role_name.into_inner(), members: vec![], rules: None };
 
 			// Check that the role does not exists already in the cluster
-			if <Clusters<T>>::get(&cluster_id)
-				.iter()
-				.any(|cluster| cluster.roles.iter().any(|role| new_role.name.eq(&role.name)))
+			if <Clusters<T>>::get(&cluster_id).ok_or(Error::<T>::ClusterNotFound)?
+				.roles.iter().any(|role| new_role.name.eq(&role.name))
 			{
 				return Err(Error::<T>::RoleExists.into())
 			}
 
 			// write
 			<Clusters<T>>::mutate(&cluster_id, |cluster| {
-				let cluster = cluster.as_mut().unwrap();
+				let cluster = cluster.as_mut().ok_or(Error::<T>::SystematicFailure).expect("Should find the cluster");
```

Comment:
ok_or(_) + expect seems off. Pick one.

Reply:
I used both here because `ok_or` returns a `Result<T, E>`, which cannot be used in a closure that returns `()`, like in this case., I normally do `ok_or(..)?`, but since here I need the `ok` value of `Result`, I thought to use `expect` .
Would this work for you?